tests/batching_test.py

```diff
@@ -366,7 +366,97 @@ def testSortKeyVal(self):
     self.assertAllClose(sk, k[:, ::-1], check_dtypes=True)
     self.assertAllClose(sv, onp.broadcast_to(v[0, ::-1], (3, 4)),
                         check_dtypes=True)
-
+  
+  def testConvGeneralDilated(self):
+    W = np.array(onp.random.randn(3, 3, 1, 5), dtype=onp.float32)
+    X = np.array(onp.random.randn(10, 5, 5, 1), dtype=onp.float32)
+
+    def f(params, x):
+      one = (1, 1)
+      dimension_numbers = ('NHWC', 'HWIO', 'NHWC')
+      y = lax.conv_general_dilated(
+          x, params, one, 'SAME', one, one, dimension_numbers)
+      return y
+    grad_loss = grad(lambda params, x: np.mean(f(params, x) ** 2))
+
+    # NOTE(schsam): We have to do this reshape thing because the current
+    # implementation of conv general expects a 4d-matrix.
```

Comment:
I might be misunderstanding, but could this comment be mistaken? `lax.conv_general_dilated` should be able to take any number of dimensions, specified either using [the string convention](https://github.com/tensorflow/tensorflow/blob/c8128f9dbebdb2327bd369cbc7ca774434b41821/tensorflow/compiler/xla/python/xla_client.py#L1460-L1478) (like e.g. `('NHWDC', 'OIHWD', 'NCHWD')`) or a [`ConvDimensionNumbers` namedtuple that is isomorphic to XLA's ConvDimensionNumbers proto](https://github.com/google/jax/blob/b23282b917953b357a12995d0fc55424d63b265c/jax/lax.py#L2957-L2970).

That doesn't mean you need to change the test code, but `conv_general_dilated` is super general, despite the fact that we haven't communicated that anywhere.

Reply:
Ah I'm sure you're right, I removed the comment! Do you think there's a more idiomatic way to write this?